viewing_party/party.py

```diff
@@ -1,23 +1,223 @@
+
+    # check each friend's watched list
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            # make sure user hasn't watched it and it's on a subscribed service
+            if (
+                movie["title"] not in user_watched_titles
+                and movie["host"] in user_data["subscriptions"]
+                and movie["title"] not in unique_recommendations  # avoid duplicates
+            ):
+                recommended_movies.append(movie)
+                unique_recommendations.append(movie["title"])  # track added movies
+
+    return recommended_movies
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    # Step 1: Count genres
+    genre_count = {}
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre in genre_count:
+            genre_count[genre] += 1
+        else:
+            genre_count[genre] = 1
+
+    # Step 2: Find the most watched genre    
+    most_watched_genre = None
+    highest_count = 0
+    
+    for genre, count in genre_count.items():
+        if count > highest_count:
+            highest_count = count
+            most_watched_genre = genre
+
+    # Step 3: Find recommended movies
+    user_watched_titles = []
+    for movie in user_data["watched"]:
+        user_watched_titles.append(movie["title"])
+
+    friend_watched_movies = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friend_watched_movies:  # Avoid duplicates
+                friend_watched_movies.append(movie)
+
+    recommended_movies = []
+    for movie in friend_watched_movies:
+        if (movie["title"] not in user_watched_titles 
+                and movie["genre"] == most_watched_genre):  
+            recommended_movies.append(movie)
+
+    return recommended_movies
+
+# wave 5.2 
+def get_rec_from_favorites(user_data):
+
+    # if user has no friends, return user's favorite movies
+    if not user_data["friends"]:
+        return user_data["favorites"]
+
+    recommended_movies = []
+
+    # go through each movie in user's favorites
+    for movie in user_data["favorites"]:
+        # assume no friend has watched it
+        movie_watched_by_friend = False
+
+        # check if any friend watched this movie
+        for friend in user_data["friends"]:
+            for watched_movie in friend["watched"]:
+                    if movie["title"] == watched_movie["title"]:
+                        movie_watched_by_friend = True
+                        break
+
+            # if a friend has watched it, move on to next favorite
+            if movie_watched_by_friend:
+                break
+
+        # if no friends watched then recommend
+        if not movie_watched_by_friend:
+            recommended_movies.append(movie)
+
+    return recommended_movies
```

Comment:
There's a fair amount of repeated logic in this function. Which existing function could you invoke here to make this more concise? It would bring this function down to 6 lines of code.